### PR TITLE
Added support for loading test results from local file

### PR DESCRIPTION
**Pull request by mateuszmidor**
Will be used for testing the webapp in CI pipeline, or for demos.

### CODE REVIEW

File: infrastructure/data_access/synthetics_repo_localfile.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+from domain.types import TestID
+
+
+class SyntheticsRepoLocalFile:
+    """
+    SyntheticsRepoLocalFile implements domain.Repo protocol.
+    It allows to load/store MeshResults to a file.
+    """
+
+    def __init__(self, mesh_test_results_filename: str) -> None:
```
Comment (Fenthick) [inline infrastructure/data_access/synthetics_repo_localfile.py:13]:
```suggestion
    def __init__(self, mesh_test_results_file_path: str) -> None:
```

This is not file name, but a path. The same with other "filename" occurrences

https://docs.python.org/3/library/functions.html?highlight=open#open
> file is a path-like object giving the pathname (absolute or relative to the current working directory) of the file to be opened or an integer file descriptor of the file to be wrapped.

File: README.md [hunk 2]
```diff
@@ -49,6 +49,12 @@ docker run \
 Configuration is stored in config file [config.yaml](./data/config.yaml)  
 UI customization is possible by modifying CSS files in [./data/assets](./data/assets)
 
+## Running the application with offline test results
+
+It is possible to load mesh test results from a local file instead of fetching from API server by specifying the file name as an environment variable:
+```bash
+export KTDATA_FILENAME=data/mesh_results_20x20.data
```
Comment (Fenthick) [inline README.md:56]:
```suggestion
export KTDATA_FILE_PATH=data/mesh_results_20x20.data
```

This is not file name, but a path.

File: main.py [hunk 3]
```diff
@@ -136,6 +136,19 @@ def get_auth_email_token() -> Tuple[str, str]:
         raise Exception(f"{err} environment variable is missing")
 
 
+def get_test_results_repo(timeout: Tuple[float, float]) -> Repo:
+    # use offline mesh test results
+    filename = os.getenv("KTDATA_FILENAME")
```
Comment (Fenthick) [inline main.py:141]:
```suggestion
    file_path = os.getenv("KTDATA_FILE_PATH")
```
Ditto.

File: main.py [hunk 4]
```diff
@@ -136,6 +136,19 @@ def get_auth_email_token() -> Tuple[str, str]:
         raise Exception(f"{err} environment variable is missing")
 
 
+def get_test_results_repo(timeout: Tuple[float, float]) -> Repo:
+    # use offline mesh test results
```
Comment (Fenthick) [inline main.py:140]: IMO redundant comments.

**Review by Fenthick — COMMENTED** [review]
Stylistic comments

**Comment by mmac-m3a** [comment]
It is good to have the test. However, I'm not a fan of having test support built into the main code path. I prefer solution based on the app fetching data from a fake API server (via a API URL config option). It is more work, but it is a higher quality test and it has zero chance to interfere with normal operations of the app.

**Comment by mateuszmidor** [comment]
OK I will rework the scenario so that webapp connects to a local apiserver that loads test data from file.

**Comment by mmac-m3a** [comment]
Replaced by https://github.com/kentik/sla_dashboard_webapp/pull/47
